ql/termstructures/yield/ratehelpers.hpp

```diff
@@ -313,6 +314,16 @@ namespace QuantLib {
                        Date customPillarDate = Date(),
                        bool endOfMonth = false,
                        const ext::optional<bool>& useIndexedCoupons = ext::nullopt);
+        SwapRateHelper(const Handle<Quote>& rate,
+                       const MakeVanillaSwap& swapBuilder,
+                       Handle<Quote> spread = {},
+                       Pillar::Choice pillar = Pillar::LastRelevantDate,
+                       Date customPillarDate = Date());
+        SwapRateHelper(Rate rate,
+                       const MakeVanillaSwap& swapBuilder,
+                       Handle<Quote> spread = {},
+                       Pillar::Choice pillar = Pillar::LastRelevantDate,
+                       Date customPillarDate = Date());
```

Comment:
There are multiple pairs of ctors which differ only in `Rate rate` vs `const Handle<Quote>& rate`. Should we try to find a way to reduce this duplication? I could imagine there might be a way to define a macro which produces both declarations in the header. The implementation in the cpp would stay as is.